### PR TITLE
refactor(codegen): simplify printing annotation comments

### PR DESCRIPTION
**Pull request by Dunqing**
Simplify printing annotation comments, reusing the architecture of printing JSDoc.

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6027?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#6027** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6027?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6021** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6021?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_codegen/tests/integration/snapshots/pure_comments.snap [hunk 1]
```diff
@@ -241,7 +241,7 @@ const defineSSRCustomElement = /* @__NO_SIDE_EFFECTS__ */ (
 };
         
 ----------
-const defineSSRCustomElement = /* #__NO_SIDE_EFFECTS__ */ (options, extraOptions) => {
+const defineSSRCustomElement = /* #__NO_SIDE_EFFECTS__ */ /* @__NO_SIDE_EFFECTS__ */ (options, extraOptions) => {
```
Comment (Dunqing) [inline crates/oxc_codegen/tests/integration/snapshots/pure_comments.snap:244]: I am not sure able to keep all annotation comments. Can you take a look this? @IWANABETHATGUY

Comment (IWANABETHATGUY) [reply crates/oxc_codegen/tests/integration/snapshots/pure_comments.snap:244]: It's just for prettier codegen, I follow the esbuild before, https://esbuild.github.io/try/#YgAwLjIwLjIALS1idW5kbGUgLS1taW5pZnk9ZmFsc2UgLS1mb3JtYXQ9ZXNtAGUAZW50cnkuanMAZXhwb3J0IGNvbnN0IGEgPSAvKiAjX19OT19TSURFX0VGRkVDVFNfXyAqLyAvKiAjX19OT19TSURFX0VGRkVDVFNfXyAqLyAoKSA9PiB7Cn0AAGZpbGUuanMAdmFyIHYgPSB0eXBlb2YgbG9jYWxWYXIKbW9kdWxlLmV4cG9ydHMgPSB2

**Review by Boshen — APPROVED** [review]

File: crates/oxc_codegen/tests/integration/snapshots/pure_comments.snap (hunk 1 above)
Comment (Dunqing) [reply crates/oxc_codegen/tests/integration/snapshots/pure_comments.snap:244]: OK, Thanks!

**Pull request merged**